spec/controllers/state_file/questions/nj_review_controller_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+
+RSpec.describe StateFile::Questions::NjReviewController do
+  let(:intake) { create :state_file_nj_intake, :df_data_minimal }
+  before do
+    sign_in intake
+  end
+
+  describe "#edit" do
+    render_views
+    it 'succeeds' do
+      get :edit
+      expect(response).to be_successful
+    end
+
+    context 'when no dependents and gross income at threshold' do
+      it 'does not show the dependents without health insurance block' do
+        allow_any_instance_of(Efile::Nj::Nj1040Calculator).to receive(:calculate_line_29).and_return 20_000
```

Comment:
I'm not sure if this is a test or app functionality but just highlighting that the income threshold is just one of the exceptions to health insurance eligibility (the other being claimed as dependents on another return) and it's 20k for some filing statuses, 10k for others

Reply:
That is a good point - instead of getting into this or that reason for eligibility, which is all tested elsewhere, the mock here can have something like:

```
allow_any_instance_of(StateFileNjIntake).to receive(:has_health_insurance_requirement_exception?).and_return(true)
allow_any_instance_of(StateFileNjIntake).to receive(:eligibility_all_members_health_insurance_no?).and_return(true)
```